lib/data-access/investments.ts

```
@@ -0,0 +1,84 @@
+        const investmentsContent = await readFile(localInvestmentsPath, 'utf-8');
+        let currentId: string | null = null;
+        // Regex to find blocks starting with 'id:'
+        const investmentBlocks = investmentsContent.split(/^\s*{\s*(?:"|')id(?:"|'):/m);
+
+        for (let i = 1; i < investmentBlocks.length; i++) { // Start from 1 as split result at 0 is before first match
+            const block = investmentBlocks[i];
+            if (!block) continue;
+
+            // Extract the ID value itself
+            const idMatch = block.match(/^(?:"|')([^"']+)(?:"|')/);
+            if (idMatch && idMatch[1]) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)